ci/docker/data.build.Dockerfile

```diff
@@ -28,6 +28,11 @@ sudo apt-get purge -y mongodb*
 sudo apt-get install -y mongodb
 sudo rm -rf /var/lib/mongodb/mongod.lock
 
+# Dependency used for read_tfrecords function.
+# Given that we only use the ExamplesToRecordBatchDecoder
+# which is purley c++, we can isntall it with --no-dependencies.
+pip install tfx-bsl==1.14.0 --no-dependencies
```

Comment:
this feels like a rather weird way to do it.. is this the intended way to use `tfx-bsl`? are there other more direct ways to import and use the logic in `ExamplesToRecordBatchDecoder`?

Reply:
other more direct ways like adding the dependency like any other dependency or are you thinking about something else?

I tried adding it as a direct dependency in the data test dependencies, but it conflicted with other existing dependencies.

We do not really need to bring transitive dependencies since the only thing we need to use is the `ExamplesToRecordBatchDecoder` which is self contained and doesn't need any extra dependency to work. Another approach we could take is to add this class to ray repo itself, I am not really familiar with that part of ray codebase but we would have to add the c class and add a python binding

Reply:
> conflicted with other existing dependencies.

could you elaborate on the specifics?

cc @can-anyscale // example that we might need multiple constraint files.

I am worrying that this is a hack, and might be not sustainable. Like is there any guarantee that when `tfx-bsl` has a new version in the future, this will still work? Do we expect users to all install `tfx-bsl` with no `--no-dependencies`

This is not a test-only dependency, but a data dependency. at the end of the days, for it to be useful, it needs to work with other dependencies in key workflows / workloads. Tests and CI are proxies to "ray works for users".

It seems that this is only used as part of the internal implementation, not part of ray data interface? if that is the case, can we fork https://github.com/tensorflow/tfx-bsl , give it another package name, remove the other parts that we do not need, and build it from source, and publish a `tfs-bsl-ray` package just for this?

Reply:
or maybe we should try to resolve the dependency conflicts..

Reply:
> > Do we expect users to all install tfx-bsl with no --no-dependencies
> 
> yes, we do

:) this is more of a question of ray code owners. I am asking @scottjlee to help here.

Reply:
I don't think we can always make the assumption (or force users to) install with the `--no-dependencies` flag, since they might be using the library's other features. But that would mean the dependency resolution would be put on the user, which we should try to avoid.